packages/gatsby/src/commands/clean.js

```diff
@@ -0,0 +1,40 @@
+const fs = require(`fs-extra`)
+const execa = require(`execa`)
+const path = require(`path`)
+
+const YARN_COMMAND = `yarnpkg`
+
+const install = async ({ directory: rootPath, report, useYarn }) => {
```

Comment:
> I should encapsulate/re-use this, since it's just copy pasted.

Yeah, maybe we should extract it as a helper since it's almost  the same as `init-starter.js` ?
https://github.com/gatsbyjs/gatsby/blob/2032147f23d40fe8d986fd33996a8c5362e6c75d/packages/gatsby-cli/src/init-starter.js#L30-L43

Reply:
might be a good idea

Reply:
The issue is that if we make changes, we need to ensure those changes are available in both gatsby/gatsby-cli. For instance, if we make this change and encapsulate a reusable helper--we are then coupling some version of gatsby with some version of gatsby-cli (depending on where we place the helper).

If you look through some of the commits, this is something @pieh noted, so it led to a small re-factor.

That said, I agree--it sucks duplicating this, but it's a little safer for now. When we're ready to make a breaking change, I think there are a lot of improvements we can make to the CLI that

Reply:
Is it feasible to put the `install` function into `create-cli.js`, then pass it down ?
https://github.com/gatsbyjs/gatsby/blob/35a0a800ff744022e7aa7cdb4fd894db0b070e3c/packages/gatsby-cli/src/create-cli.js#L94

```diff
- let args = { ...argv, ...siteInfo, useYarn }
+ let args = { ...argv, ...siteInfo, useYarn, install }
```

Reply:
Yes - except that's a bit strange and I honestly prefer duplicating here.

Also then this breaks if gatsby-cli is < the version where this feature was introduced (which is a pretty common use case?)

Reply:
Let's get this PR move forward👍.